Slightly better instructions for my setup which seems to be a bit different from the one expected  by the documentation.

djfm opened the pull request:
Mostly Python `venv` madness... I don't think this can induce regressions and it may help a few people. Python is black magic to me sorry if that is actually a non issue and I'm the only Python illiterate programmer in the world 🤣.

djfm commented on classes/mainwindow.py:
This used to crash randomly. Superficial fix.

djfm commented:
Now I can start the tool on a channel and make it download all vids in one go, which it failed to do before

hyperfield commented:
Thank you for your contributions! Could you please update your commit messages to follow best practices by making them more descriptive and easier to understand? That would help better understand your changes in context and ensure the repository stays up to standard. For tips on writing effective commit messages, you can refer to [this guide](https://cbea.ms/git-commit/). If you need help changing existing commit messages, [this documentation](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/changing-a-commit-message) provides step-by-step instructions.

hyperfield commented:
@djfm Thank you for your contributions! Could you please update your commit messages to follow best practices by making them more descriptive and easier to understand? That would help better understand your changes in context and ensure the repository stays up to standard. For tips on writing effective commit messages, you can refer to [this guide](https://cbea.ms/git-commit/). If you need help changing existing commit messages, [this documentation](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/changing-a-commit-message) provides step-by-step instructions.

hyperfield commented:
No worries. I'll go over your commits and review them as soon as possible. I might need to tweak some of the commit messages, though. Again, thanks for your contribution. I’ve already seen the code you fixed in the app, and I think it makes good sense.

djfm commented:
Thanks, sorry I can't be of that much help.

I realize I indeed haven't been as clear as I thought I was in my PR
description and messages.

There are 3 aspects in my PR:

- main issue: for at least 2 different reasons sometimes the program
crashes, because it tries to access some property on an object that's
sometimes missing, and for another reason that I forgot

- secondary issue: I'm not very familiar with the python ecosystem but on
my system (Ubuntu 24.04 LTS) python virtual environments do not expose any
script named `activate` so I added a script which prepends the binary
directory of the virtual env to the path before invoking python to launch
the app, thus using the correct python version regardless of weather the
`activate` script is present or not

- minor issue: this one is about style and linting, I'm a bit of an
extremist regarding linting, so I committed my vscode settings (but I
realize there's a couple that shouldn't be there) and a bunch of fıles
changed automatically just because I saved them and the linters /
formatters did their job. I tend to commit vscode settings because I think
it's useful if everyone uses the same or as close as possible but the
practice is debatable I'll grant you that

As I said I'm a little bit short on time so I really did the quickest fix I
could come up with and I did not spend much time trying to figure out the
logic of the app.

Basically what I'm doing is just starting download threads again if an
exception is caught and there's still stuff to process.

My fork seems to be working as intended (by the both of us ;)) as I've
successfully downloaded what really seems to be the entirety of the videos
of two channels, and the program doesn't enter an infinite loop once it's
done downloading, which was a fear I had at some point.

So this is not a very smart fix. I think it's an improvement and I think it
works, but there is probably a much cleaner way to fix the issue and there
are probably edge cases I'm not aware of.

You can even totally discard the PR and fix the issues yourself more
cleanly and it probably won't take you very long as you know the code.


On Sat, Apr 12, 2025, 14:44 hyperfield ***@***.***> wrote:

> No worries. I'll go over your commits and review them as soon as possible.
> I might need to tweak some of the commit messages, though. Again, thanks
> for your contribution. I’ve already seen some of the code you fixed in the
> app, and I think it makes good sense.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hyperfield/yt-channel-downloader/pull/31#issuecomment-2798818095>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AALESE5WPPLAA3JVVE3DL6T2ZEDCZAVCNFSM6AAAAAB2M2VXIGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOOJYHAYTQMBZGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
> *hyperfield* left a comment (hyperfield/yt-channel-downloader#31)
> <https://github.com/hyperfield/yt-channel-downloader/pull/31#issuecomment-2798818095>
>
> No worries. I'll go over your commits and review them as soon as possible.
> I might need to tweak some of the commit messages, though. Again, thanks
> for your contribution. I’ve already seen some of the code you fixed in the
> app, and I think it makes good sense.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hyperfield/yt-channel-downloader/pull/31#issuecomment-2798818095>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AALESE5WPPLAA3JVVE3DL6T2ZEDCZAVCNFSM6AAAAAB2M2VXIGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOOJYHAYTQMBZGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

hyperfield commented:
PR superseded by #32 

I have now merged your contributions into the `main` branch. I dropped some commits and merged your bug fixes, and I changed the commits' descriptions to more precise. I think your scripts idea is interesting, and I'm going to implement a Python script for automating the creation of a virtual environment and installing any needed dependencies. (Also feel free to implement it yourself if you want.) As a side note, `activate` should definitely be present in your virtual environment, if you're on Linux or Mac, or it's activate.bat on Windows.
